### DISCUSSION TITLE
New Greek arrives

### DISCUSSION THREAD

**Discussion by psb1558** [psb1558/Junicode-font#84] [General]
I now have a basic lowercase Greek set for Junicode 2. It needs work, but it strikes me as not too bad, at least for when Greek text needs to be quoted in a Latin context or for the odd mathematical expression.
<img width="489" alt="image" src="https://user-images.githubusercontent.com/16215767/141653572-d502bf35-4a0d-4e01-8659-89a582accc48.png">
My aim was to adopt a style that would harmonize well with  Latin script: users of GFS Elpis may notice the influence of that font. But weight and proportion and many details are pure Junicode.
There is also a small cap set, matching the small caps of Junicode:
<img width="533" alt="image" src="https://user-images.githubusercontent.com/16215767/141653619-714d21b2-28be-4e84-8293-646e248aab42.png">
The full Unicode polytonic set is supported in lowercase, uppercase and small caps, but so far I have added little programming to support it. I welcome comments from any who know Greek script better than I do (that would be almost everyone).

The new Greek set is in version 1.028 beta. Roman only: the italic still has the old-style Wilson set, which has a sort of italic look.

**Comment by Nuzzinamo** [comment]
Dear Prof. Baker:
In MS Word (Professional Plus 2021), Greek words betwixt Latin (or other) ones are automatically (forcibly) changed to (apparent) small caps against my will in nearly, if not exceptionlessly all cases (I simply can’t ﬁgure out any exact conditions for this, although sometimes the effect seems to be thwartable [i. e., true lowercase restituted])—
Now the truly frustrating thing is that the (forced, undesired) small caps do not appear *as* such, don’t appear “styled” in Word’s font menu—so I just can’t undo them!
This happens only with Junicode (at my place:  Two Beta);  in my example attached, I have had to resort to the Brill font for most of the Greek (which isn’t desirable for me either)—yet what went (and [nearly] always, at any rate uncontrollably does go) wrong with the Junicode strings, you can discern in the “δόγμᾰ … δόξᾰ„ line.
(… This text is NOT ‘styled‘ small caps or otherwise!).
Since I need my paper for a talk I’m supposed to give at the 34th Deutscher Orientalisten-Tag (DOT) in Berlin (mid) NEXT MONTH,
your eventual help should greatly be appreciated.
Yours sincerely,
Andreas Ehlers,
Hamburg University.

![Junicode Greek today](https://user-images.githubusercontent.com/111339604/184818051-86b50916-7adc-461f-9d8d-f6dff2c9372f.png)

**Reply by psb1558** [reply]
I've written to you separately asking for a minimal MS Word doc that shows the problem.

**Reply by Nuzzinamo** [reply]
I shall be sending it to you in an instant!

**Comment by psb1558** [comment]
When I open the document in Word for Mac (version 16.60), I do not see the intrusive small caps, but I do see the misaligned diacritics. I also see the misaligned diacritics when I open in LibreOffice, which means that more than one layout engine is producing the effect. When I type a sequence myself (as opposed to just opening your file), the diacritics are aligned correctly:
<img width="266" alt="image" src="https://user-images.githubusercontent.com/16215767/185112530-0964117e-603d-4721-9679-062335ee225d.png">
Likewise if I "sanitize" a sequence by copying it from your document, pasting it into a plain text editor, and pasting it back into the Word document, the diacritics end up correctly aligned:
<img width="484" alt="image" src="https://user-images.githubusercontent.com/16215767/185113120-fae60d08-e5b0-469d-950a-80ded83f4481.png">
Okay, I've got it: the language setting is changing in mid-word, probably between base character and diacritic. This is not allowed, as any formatting characters between a base character and a combining mark will prevent the correct positioning of the mark. Select your Greek passages and set the language to Greek (or German, if you prefer, for individual words and very short Greek passages—only make sure the language doesn't change mid-word), and the diacritics should line up.

I'm going to be away from my computer for a couple of days. I'll investigate the small cap problem further when I get back. Meanwhile, check to see if tidying up the language marking in your document makes a difference.

I don't understand what the blue and green highlighting is trying to tell me. For correct ligatures, make sure "Contextual Alternates" are turned on in the "Advanced" tab of the "Font" dialog. (In fact, "Contextual Alternates" and "Standard Ligatures" should always be on for all your documents. It was a horrible decision by Microsoft to have these features off by default.)

**Comment by psb1558** [comment]
Also: opening up the docx file and looking inside shows that there are not only language changes inside of words, but a real tangle of formatting marks, a lot of which I don't understand. But cleaning these up (try that eraser  button in the "Font" ribbon, and if that fails, paste into a plain text editor then back into Word) may possibly fix more problems than the diacritic alignment.

**Comment by psb1558** [comment]
To explain further, here is the under-the-hood markup (made "pretty" for easy reading) that produces 
<img width="170" alt="image" src="https://user-images.githubusercontent.com/16215767/185515073-3e2cd6f5-4ce4-4bcd-8b58-9b9f29545f69.png">
```
      <w:r w:rsidR="00792B82" w:rsidRPr="00AD6A98">
        <w:rPr>
          <w:highlight w:val="yellow"/>
          <w:lang w:val="el-GR"/>
        </w:rPr>
        <w:t>α</w:t>
      </w:r>
      <w:r w:rsidR="00792B82" w:rsidRPr="00AD6A98">
        <w:rPr>
          <w:highlight w:val="yellow"/>
        </w:rPr>
        <w:t>̆̓</w:t>
      </w:r>
      <w:r w:rsidR="00792B82">
        <w:rPr>
          <w:lang w:val="el-GR"/>
        </w:rPr>
        <w:t>μᾰθ</w:t>
      </w:r>
      <w:r w:rsidR="00792B82" w:rsidRPr="00325C62">
        <w:rPr>
          <w:highlight w:val="yellow"/>
          <w:lang w:val="el-GR"/>
        </w:rPr>
        <w:t>ι</w:t>
      </w:r>
      <w:r w:rsidR="00792B82" w:rsidRPr="00325C62">
        <w:rPr>
          <w:highlight w:val="yellow"/>
        </w:rPr>
        <w:t>̆́</w:t>
      </w:r>
      <w:r w:rsidR="00792B82" w:rsidRPr="0062046E">
        <w:rPr>
          <w:color w:val="C40233"/>
          <w:lang w:val="el-GR"/>
        </w:rPr>
        <w:t>ᾱ</w:t>
      </w:r>
```
If you hunt around in there, you will spot your Greek characters (text falls between `<w:t></w:t>` tags). Initial alpha isolated in its own `<w:t></w:t>` tags, the diacritics that are supposed to be positioned on top of it in another "run" of text tags. This litter of extra tags seems to be a by-product of intensive editing in Word, which, remarkably, seems to do no cleanup of messy markup (as LibreOffice Writer does). Again, base characters must never be separated by markup from their diacritics.

By contrast, look at the markup for the German text that immediately precedes that Greek word:
```
<w:t xml:space="preserve">„ein unuͤglicher Habitus vernuͤnigen Handelns ⟨iſt⟩ in Dingen, die fuͤr den Menſchen Guͤter und Uͤbel nd“ </w:t>
```
Very clean! (only Junicode's PUA characters can't be displayed here).

Everywhere you see this problem in the document, e.g. in Brill
<img width="74" alt="image" src="https://user-images.githubusercontent.com/16215767/185517758-8f76327e-0203-41ce-af3e-58991eaa3f3a.png">
you have the same problem of intrusive markup. I don't know a lot about Word (I don't use it because of its limitations), but it *might* work simply to change the formatting of a Greek passage (e.g. applying italics and then changing back to roman); but the absolutely reliable nuclear option is to copy the text out of the Word document, paste it into a plain text editor (maybe Notepad in Windows or TextEdit on the Mac), then copy it out of the plain text editor and paste it back into the Word document.

Once again I haven't been able to reproduce the small cap problem in my copy of Word, and I don't see it when I open the test file in other programs (e.g. LibreOffice Writer and Pages). But I have figured out from your image that what's being displayed there is not Junicode's set of small caps at all. Instead, what you're seeing are Junicode's standard capitals scaled to small-cap size.

It turns out that Word is notorious for never using the small caps that are built into so many modern fonts (including Junicode and Brill), but instead faking small caps by scaling the font's standard capitals. So, since Junicode does not produce small caps by scaling the capitals, the problem you're seeing is almost certainly *not* a bug in Junicode, but a bug in your particular version of Word, probably triggered by the complexity of the markup. (Notice how the ᾱ in the word for which I've shown the markup above, and which displays for you as small cap, is isolated in its own set of text tags. Word seems to be misinterpreting the markup there as an instruction to turn on small caps for that letter.) I'd suggest using the cleanup procedure outlined above to simplify the markup and see if that fixes the small cap problem as it fixes the diacritic problem.

This is a rather long explanation, and I hope not too technical. I must say that Word's miserable font handling is a big frustration to me. LibreOffice Writer does much better and will export a Word document for you when it's time to send your document off to your publisher.

**Comment by Nuzzinamo** [comment]
Dear Professor Baker!

Thank you so much for your answer—it has already done some good to my work.

▶ Can you, please, help me 1 step farther .?:

For various reasons, I need to use “Word” … please look how many languages and typefaces I’m forced to operate with [cf. append.] …

Now, _how can I make Your_ (desirable) _“Junicode” LIGATURES_
work in “Word”:  so as for them to be “**exported**” duly **→** “**p. d. f.**”?

It’s really near-to impossible to work around this, e. g. by switching to another text processor …
If you can help, _please_ do it … and it would be a great beneﬁt to “Junicode,” too, wouldn’t it,
if it hadn’t just to ‘stop’ before “Word”—.

Looking forward very much to Your eventual reply!
My preparations for that talk in Berlin need to be **completed** ᴛʜıs sᴀᴛᴜʀᴅᴀʏ  . . .

Best regards:
Andreas Ehlers.
University of Hamburg,
Germany.

![MS_Word_today](https://user-images.githubusercontent.com/111339604/188320124-5ed5f148-17ba-4b4c-b80c-0de10be3b836.png)

**Comment by psb1558** [comment]
Word doesn't turn on ligatures by default: you have to enable them explicitly. (Best to do this with styles, if you can, and not by turning on these features for every passage that needs them.)

In the "Font" dialog, click over to the "Advanced" tab and check the appropriate boxes:
![image](https://user-images.githubusercontent.com/16215767/188320857-c6d9891b-60d4-42bd-8454-3cb937ad9ce1.png)

Note: The "kerning" box **must** be checked: for some reason this enables all the "Advanced Typography." Because of the way Junicode does ligatures, **both** the "ligatures" **and** the "contextual alternates" boxes must be checked. (You can also choose the number style you like, and use Junicode's stylistic sets—but only one stylistic set at a time (other, less brain-dead applications allow you to apply several).

Be sure to download the current version. The kerning program I was using messed up the ligatures a few versions back, but I have now repaired them, and also worked out a few bugs. If you have any problems with specific combinations, let me know: I should be able to repair them quickly.

Or am I missing other problems?

**Reply by Nuzzinamo** [reply]
Dear Professor Baker,
please see the problem on the screenshot beneath.
It appears only WHEN EXPORTED TO "P. D. F.”
(the LIGATURES don’t show as they should).
Thank you so much for your help!
It’s truly instrumental.  I bow to your kindness!

Sincerely,
Andreas Ehlers.

![MS_Word_today_(LIGATURES)](https://user-images.githubusercontent.com/111339604/188335918-db48d981-d23a-4d9b-b4eb-515f8556c00c.png)

**Comment by psb1558** [comment]
Well, this looks very bad! I can't think what in the font could be triggering this behavior, and I've had no other reports of it. I've got several suggestions:

- If you're using the ttf version of Junicode, try the otf version (or vice versa). Make sure to uninstall the old version before trying the new.
- Try a different method of exporting the PDF. If you're doing File-->Export, try "Save As" (choosing PDF as the save format), or vice versa.
- Try an external program for exporting to PDF (Acrobat, or maybe the freeware [CutePDF](https://www.cutepdf.com/index.htm) or [PrimoPDF](https://www.gonitro.com/primopdf)).
- I've built the fonts by an alternative method, [here](https://github.com/psb1558/Junicode-font/blob/master/fonts/junicode_alternate_build.zip). See if these make a difference.

Be sure to let me know if the first or last of these work for you, because that will mean that I have to track down a bug (or post a bug report for the font-building software I'm using).

**Comment by Nuzzinamo** [comment]
Thank you, I shall try this!
Btw, the kerning with f (but also ſ, for ſure) + ’, ‘, b, h, &c. is not good – & there’s no “ﬀ” in Junicode Two Beta Condensed, _Italics_ (though there’s a true ligature “ſſ”!).
… more to come once I’ll have found out more!
Yours, A. E.

**Comment by Doc73** [comment]
Dear Professor,
I really like your font and I'm seriously considering using it for my book, a critical edition of a Greek text.
Just one question: are you planning to update the italic (and bold italic) shape? If yes, when will be available?

Unfortunately, I don't know how to use programs to create fonts, such as Fontlab, but, in what I can, I would gladly make some contributions on my part.

Many thanks in advance,
DC

**Reply by psb1558** [reply]
I've been working on this font for over thirty years. I keep plugging along. Eventually, either the font will be done, or I will.
